docs/deliverables-portal.md

```diff
@@ -0,0 +1,134 @@
+                project_case.cases.forEach((donor_case) => {
+                    deliverable.deliverables.forEach((deliverables) => {
+                        tableRows.push(tableRow(null,
+                            {contents: "testDummyData"},
+                            {contents: "testDummyData"},
+                            {contents: "testDummyData"},
+                            {contents: "testDummyData"},
+                            {contents: "testDummyData"}))
+                      })
+                })
+            })
+        }); 
+
+    return tableRows; 
+}
+````
+
+2. Function to instantiate the deliverables table, using the bootstrapTable function in `html.ts`. This function also takes the `deliverablesObject` as a parameter. This function is also where the table headers and rows are appended to the table itself.
+````
+export function deliverablesTable(deliverables: DeliverableObject[]): HTMLElement {
+    const pageContainer = document.createElement("div");
+    const pageHeader = document.createElement("h3");
+    pageHeader.innerText = "Deliverables Portal";
+
+    const tableHeaders = new Map([
+        ["project_id", "Project ID"],
+        ["case_id", "Case ID"],
+        ["location", "Location"],
+        ["notes", "Notes"],
+        ["expiry_date", "Expiry Date"]]);
+
+    const table = bootstrapTable(tableHeaders, null, "table", true, true, true, true);
+    const tableRows = constructTableRows(deliverables);
+    const tableBody = tableBodyFromRows(null, tableRows);
+    table.appendChild(tableBody);
+    table.className = "deliverables-table";
+    pageContainer.appendChild(pageHeader);
+    pageContainer.appendChild(table);
+    document.body.appendChild(pageContainer);
+
+    return pageContainer;
+}
+````
+
+3. Function to initialize the deliverables. This function is primarily made up of jQuery code and is where the deliverables data is fetched from the API. 
+````
+export function initialiseDeliverables() {
+    const closeBusy = busyDialog();
+
+    Promise.all([
+        fetch("api/deliverables")
+    ])
+        .then(responses => {
+            return Promise.all(responses.map(response => response.json()));
+        })
+        .then((responses) => {
+        const deliverables = responses[0] as DeliverableObject[]
+        document.body.appendChild(deliverablesTable(deliverables));
+        return deliverables;
+    })
+        .then(() => {
+
+            $(function () {
+                $('#deliverables').bootstrapTable({});
+                $('#table').bootstrapTable({
+                    exportDataType: 'all'
+                });
+            })
+        })
+        .catch((error) => {
+            console.log(error); //todo: write to actual logs
+        })
+        .finally(closeBusy);
+}
+````
+
+- **deliverables.js**: JavaScript file which is automatically updated with changes made to `deliverables.ts` by running `tsc -p sampuru-ui && PATH=$(npm bin):$PATH rollup -c`.
```

Comment:
If the developer won't be touching this file directly it might be confusing to document it as if it's available for editing